ReactAndroid/src/main/java/com/facebook/react/ReactInstanceManagerBuilder.java

```diff
@@ -345,40 +356,71 @@ public ReactInstanceManager build() {
+    // So we can use the value of enableHermes receved by the constructor
+    // to decide which library to load at launch
+    if(jscOrHermes.isEmpty()) {
+      try {
+        // If JSC is included, use it as normal
+        initializeSoLoaderIfNecessary(applicationContext);
+        JSCExecutor.loadLibrary();
+        return new JSCExecutorFactory(appName, deviceName);
+      } catch (UnsatisfiedLinkError jscE) {
+        // https://github.com/facebook/hermes/issues/78 shows that
+        // people who aren't trying to use Hermes are having issues.
+        // https://github.com/facebook/react-native/issues/25923#issuecomment-554295179
+        // includes the actual JSC error in at least one case.
+        //
+        // So, if "__cxa_bad_typeid" shows up in the jscE exception
+        // message, then we will assume that's the failure and just
+        // throw now.
+
+        if (jscE.getMessage().contains("__cxa_bad_typeid")) {
+          throw jscE;
+        }
+
+        // Otherwise use Hermes
+        try {
+          HermesExecutor.loadLibrary();
+          return new HermesExecutorFactory();
+        } catch (UnsatisfiedLinkError hermesE) {
+          // If we get here, either this is a JSC build, and of course
+          // Hermes failed (since it's not in the APK), or it's a Hermes
+          // build, and Hermes had a problem.
+
+          // We suspect this is a JSC issue (it's the default), so we
+          // will throw that exception, but we will print hermesE first,
+          // since it could be a Hermes issue and we don't want to
+          // swallow that.
+          hermesE.printStackTrace();
+          throw jscE;
+        }
       }
-
-      // Otherwise use Hermes
+    }
+    // if the app explicitly specifies which engine to use, directly load
+    // that library
+    else{
       try {
-        HermesExecutor.loadLibrary();
-        return new HermesExecutorFactory();
-      } catch (UnsatisfiedLinkError hermesE) {
-        // If we get here, either this is a JSC build, and of course
-        // Hermes failed (since it's not in the APK), or it's a Hermes
-        // build, and Hermes had a problem.
-
-        // We suspect this is a JSC issue (it's the default), so we
-        // will throw that exception, but we will print hermesE first,
-        // since it could be a Hermes issue and we don't want to
-        // swallow that.
-        hermesE.printStackTrace();
-        throw jscE;
+        initializeSoLoaderIfNecessary(applicationContext);
+        if(jscOrHermes.equals("hermes")){
+          HermesExecutor.loadLibrary();
+          return new HermesExecutorFactory();
+        }
+        else{
+          JSCExecutor.loadLibrary();
+          return new JSCExecutorFactory(appName, deviceName);
+        }
+      } catch (UnsatisfiedLinkError e) {
+        // linking library failed, so we print its stacktrace and
+        // throw the exception
+        e.printStackTrace();
+        throw e;
```

Comment:
Let's try to refactor this a bit:

```
if (enableHermes == null || enableHermes == false) {
   JSCExecutor.loadLibrary();
else if (enableHermes == null || enableHermes == true) {
   HermesExecutor.loadLibrary();
}
```

There is no need to duplicate code

Reply:
This seems fine, however, I seem to have lost the original repository from where I raised the PR. I would have to raise this PR again. I will ping the link to that PR here before closing this one.

Reply:
Sure take your time 👍

Reply:
Hey @cortinico. I have raised a new and updated PR for the same here: https://github.com/facebook/react-native/pull/33952
Kindly review that.

I will close this PR once that one is reviewed.